diff -Nur gnome-panel-2.15.91/gnome-panel/panel-menu-items.c gnome-panel-2.15.91.new/gnome-panel/panel-menu-items.c
--- gnome-panel-2.15.91/gnome-panel/panel-menu-items.c	2006-08-07 23:16:07.000000000 +0200
+++ gnome-panel-2.15.91.new/gnome-panel/panel-menu-items.c	2006-08-08 09:51:04.000000000 +0200
@@ -555,6 +555,7 @@
 	GtkWidget *places_menu;
 	GtkWidget *item;
 	char      *gconf_name;
+	char      *path;
 
 	places_menu = panel_create_menu ();
 
@@ -567,6 +568,16 @@
 	if (gconf_name)
 		g_free (gconf_name);
 
+	path = g_build_filename (g_get_home_dir (), "Documents", NULL);
+	if (g_file_test (path, G_FILE_TEST_IS_DIR))
+	  panel_menu_items_append_place_item ("gnome-fs-directory",
+					      _("Documents"),
+					      _("Documents directory"),
+					      places_menu,
+					      G_CALLBACK (activate_uri),
+					      "Documents");
+	g_free (path);
+
 	if (!gconf_client_get_bool (panel_gconf_get_client (),
 				    DESKTOP_IS_HOME_DIR_KEY,
 				    NULL))
@@ -703,6 +714,10 @@
 	panel_menu_items_append_from_desktop (menu, "yelp.desktop", NULL);
 	panel_menu_items_append_from_desktop (menu, "gnome-about.desktop", NULL);
 
+	if (g_file_test (DATADIR"/omf/about-ubuntu/about-ubuntu-C.omf",
+			 G_FILE_TEST_IS_REGULAR))
+		panel_menu_items_append_from_desktop (menu, "ubuntu-about.desktop", NULL);
+
 	if (parent->priv->append_lock_logout)
 		panel_menu_items_append_lock_logout (menu);
 }
