diff -Nur gnome-panel-2.11.3.orig/gnome-panel/menu.c gnome-panel-2.11.3/gnome-panel/menu.c
--- gnome-panel-2.11.3.orig/gnome-panel/menu.c	2005-06-06 21:54:00.000000000 +0200
+++ gnome-panel-2.11.3/gnome-panel/menu.c	2005-06-27 15:21:40.770711424 +0200
@@ -1386,7 +1386,7 @@
 	return menuitem;
 }
 
-static void
+void
 create_submenu (GtkWidget          *menu,
 		GMenuTreeDirectory *directory,
 		GMenuTreeDirectory *alias_directory)
diff -Nur gnome-panel-2.11.3.orig/gnome-panel/menu.h gnome-panel-2.11.3/gnome-panel/menu.h
--- gnome-panel-2.11.3.orig/gnome-panel/menu.h	2005-04-11 16:58:41.000000000 +0200
+++ gnome-panel-2.11.3/gnome-panel/menu.h	2005-06-27 15:21:40.770711424 +0200
@@ -67,6 +67,9 @@
 gboolean menu_dummy_button_press_event (GtkWidget      *menuitem,
 					GdkEventButton *event);
 
+void create_submenu (GtkWidget          *menu,
+		     GMenuTreeDirectory *directory,
+		     GMenuTreeDirectory *alias_directory);
 
 G_END_DECLS
 
diff -Nur gnome-panel-2.11.3.orig/gnome-panel/panel-menu-items.c gnome-panel-2.11.3/gnome-panel/panel-menu-items.c
--- gnome-panel-2.11.3.orig/gnome-panel/panel-menu-items.c	2005-06-06 23:51:27.000000000 +0200
+++ gnome-panel-2.11.3/gnome-panel/panel-menu-items.c	2005-06-27 15:25:37.614705664 +0200
@@ -75,6 +75,8 @@
 	GtkWidget   *menu;
 	PanelWidget *panel;
 
+	GMenuTree    *tree;
+
 	guint        use_image : 1;
 	guint        append_lock_logout : 1;
 };
@@ -84,6 +86,8 @@
 
 static GnomeVFSVolumeMonitor *volume_monitor = NULL;
 
+static void panel_desktop_menu_item_recreate_menu (PanelDesktopMenuItem *desktop_item);
+
 static void
 activate_uri (GtkWidget *menuitem,
 	      char      *path)
@@ -607,16 +611,63 @@
 		panel_menu_items_append_lock_logout (menu);
 }
 
+static void
+handle_menu_tree_changed (GMenuTree  *tree,
+			  PanelDesktopMenuItem *desktop_item)
+{
+	panel_desktop_menu_item_recreate_menu (desktop_item);
+}
+
 static GtkWidget *
 panel_desktop_menu_item_create_menu (PanelDesktopMenuItem *desktop_item)
 {
 	GtkWidget *desktop_menu;
+	GMenuTreeDirectory *directory;
+
+	desktop_menu = create_empty_menu ();
+
+	/*
+	g_object_set_data_full (G_OBJECT (desktop_menu),
+				"panel-menu-tree-filename",
+				g_strdup ("settings.menu"),
+				(GDestroyNotify) g_free);
+
+	g_object_set_data_full (G_OBJECT (desktop_menu),
+				"panel-menu-tree",
+				menu_tree_ref (tree),
+				(GDestroyNotify) menu_tree_unref);
+
+	g_object_set_data_full (G_OBJECT (desktop_menu),
+				"panel-menu-tree-path",
+				g_strdup ("/"),
+				(GDestroyNotify) g_free);
+
+	g_object_set_data (G_OBJECT (desktop_menu),
+			   "panel-menu-needs-loading",
+			   GUINT_TO_POINTER (TRUE));
+	*/
 
-	desktop_menu = create_applications_menu ("settings.menu", NULL);
+	g_object_set_data (G_OBJECT (desktop_menu),
+			   "panel-menu-needs-appending",
+			   GUINT_TO_POINTER (TRUE));
 
-	g_signal_connect (desktop_menu, "show",
-			  G_CALLBACK (panel_desktop_menu_item_append_menu),
-			  desktop_item);
+	directory = gmenu_tree_get_directory_from_path (desktop_item->priv->tree,
+						       "/Preferences");
+
+	if (directory) {
+		create_submenu (desktop_menu, directory, NULL);
+		gmenu_tree_item_unref (directory);
+	}
+
+	directory = gmenu_tree_get_directory_from_path (desktop_item->priv->tree,
+						       "/Administration");
+
+	if (directory) {
+		create_submenu (desktop_menu, directory, NULL);
+		gmenu_tree_item_unref (directory);
+	}
+
+	panel_desktop_menu_item_append_menu (desktop_menu, desktop_item);
 
 	return desktop_menu;
 }
@@ -674,6 +725,14 @@
 		panel_lockdown_notify_remove (G_CALLBACK (panel_desktop_menu_item_recreate_menu),
 					      menuitem);
 	desktop_parent_class->finalize (object);
+
+	if (menuitem->priv->tree) {
+		gmenu_tree_remove_monitor (menuitem->priv->tree,
+					  (GMenuTreeChangedFunc) handle_menu_tree_changed,
+					  menuitem);
+		gmenu_tree_unref (menuitem->priv->tree);
+	}
+	menuitem->priv->tree = NULL;
 }
 
 static void
@@ -736,6 +795,12 @@
 				       PanelDesktopMenuItemClass *klass)
 {
 	menuitem->priv = PANEL_DESKTOP_MENU_ITEM_GET_PRIVATE (menuitem);
+
+	menuitem->priv->tree = gmenu_tree_lookup ("settings.menu", GMENU_TREE_FLAGS_NONE);
+
+	gmenu_tree_add_monitor (menuitem->priv->tree,
+			       (GMenuTreeChangedFunc) handle_menu_tree_changed,
+			       menuitem);
 }
 
 static void
