diff -Nur gnome-panel-2.15.91/applets/clock/GNOME_ClockApplet_Factory.server.in.in gnome-panel-2.15.91.new/applets/clock/GNOME_ClockApplet_Factory.server.in.in
--- gnome-panel-2.15.91/applets/clock/GNOME_ClockApplet_Factory.server.in.in	2006-08-08 01:55:52.000000000 +0200
+++ gnome-panel-2.15.91.new/applets/clock/GNOME_ClockApplet_Factory.server.in.in	2006-08-08 09:55:36.000000000 +0200
@@ -24,6 +24,7 @@
 	</oaf_attribute>
 	<oaf_attribute name="name" type="string" _value="Clock"/>
 	<oaf_attribute name="description" type="string" _value="Get the current time and date"/>
+	<oaf_attribute name="panel:category" type="string" _value="Accessories"/>
 	<oaf_attribute name="panel:icon" type="string" value="gnome-panel-clock"/>
 	<oaf_attribute name="bugzilla:bugzilla" type="string" value="GNOME"/>
 	<oaf_attribute name="bugzilla:product" type="string" value="gnome-panel"/>
diff -Nur gnome-panel-2.15.91/applets/fish/GNOME_FishApplet_Factory.server.in.in gnome-panel-2.15.91.new/applets/fish/GNOME_FishApplet_Factory.server.in.in
--- gnome-panel-2.15.91/applets/fish/GNOME_FishApplet_Factory.server.in.in	2006-08-08 01:55:52.000000000 +0200
+++ gnome-panel-2.15.91.new/applets/fish/GNOME_FishApplet_Factory.server.in.in	2006-08-08 09:56:07.000000000 +0200
@@ -24,6 +24,7 @@
 	</oaf_attribute>
 	<oaf_attribute name="name" type="string" _value="Fish"/>
 	<oaf_attribute name="description" type="string" _value="Display a swimming fish or another animated creature"/>
+	<oaf_attribute name="panel:category" type="string" _value="Accessories"/>
 	<oaf_attribute name="panel:icon" type="string" value="gnome-panel-fish"/>
 	<oaf_attribute name="bugzilla:bugzilla" type="string" value="GNOME"/>
         <oaf_attribute name="bugzilla:product" type="string" value="gnome-panel"/>
diff -Nur gnome-panel-2.15.91/applets/notification_area/GNOME_NotificationAreaApplet.server.in.in gnome-panel-2.15.91.new/applets/notification_area/GNOME_NotificationAreaApplet.server.in.in
--- gnome-panel-2.15.91/applets/notification_area/GNOME_NotificationAreaApplet.server.in.in	2006-08-08 01:55:53.000000000 +0200
+++ gnome-panel-2.15.91.new/applets/notification_area/GNOME_NotificationAreaApplet.server.in.in	2006-08-08 09:56:36.000000000 +0200
@@ -24,6 +24,7 @@
 	</oaf_attribute>
 	<oaf_attribute name="name" type="string" _value="Notification Area"/>
 	<oaf_attribute name="description" type="string" _value="Area where notification icons appear"/>
+	<oaf_attribute name="panel:category" type="string" _value="Utilities"/>
 	<oaf_attribute name="panel:icon" type="string" value="gnome-panel-notification-area"/>
 	<oaf_attribute name="bugzilla:bugzilla" type="string" value="GNOME"/>
 	<oaf_attribute name="bugzilla:product" type="string" value="gnome-panel"/>
diff -Nur gnome-panel-2.15.91/applets/wncklet/GNOME_Wncklet_Factory.server.in.in gnome-panel-2.15.91.new/applets/wncklet/GNOME_Wncklet_Factory.server.in.in
--- gnome-panel-2.15.91/applets/wncklet/GNOME_Wncklet_Factory.server.in.in	2006-08-08 01:55:53.000000000 +0200
+++ gnome-panel-2.15.91.new/applets/wncklet/GNOME_Wncklet_Factory.server.in.in	2006-08-08 09:58:27.000000000 +0200
@@ -25,6 +25,7 @@
 	<oaf_attribute name="name" type="string" _value="Window Selector"/>
 	<oaf_attribute name="description" type="string" _value="Switch between open windows using a menu"/>
 	<oaf_attribute name="panel:icon" type="string" value="gnome-panel-window-menu"/>
+	<oaf_attribute name="panel:category" type="string" _value="Desktop &amp; Windows"/>
 	<oaf_attribute name="bugzilla:bugzilla" type="string" value="GNOME"/>
 	<oaf_attribute name="bugzilla:product" type="string" value="gnome-panel"/>
 	<oaf_attribute name="bugzilla:component" type="string" value="window selector"/>
@@ -64,6 +65,7 @@
 	<oaf_attribute name="name" type="string" _value="Workspace Switcher"/>
 	<oaf_attribute name="description" type="string" _value="Switch between workspaces"/>
 	<oaf_attribute name="panel:icon" type="string" value="gnome-panel-workspace-switcher"/>
+	<oaf_attribute name="panel:category" type="string" _value="Desktop &amp; Windows"/>
 	<oaf_attribute name="bugzilla:bugzilla" type="string" value="GNOME"/>
 	<oaf_attribute name="bugzilla:product" type="string" value="gnome-panel"/>
 	<oaf_attribute name="bugzilla:component" type="string" value="workspace switcher"/>
@@ -102,6 +104,7 @@
 	</oaf_attribute>
 	<oaf_attribute name="name" type="string" _value="Window List"/>
 	<oaf_attribute name="description" type="string" _value="Switch between open windows using buttons"/>
+	<oaf_attribute name="panel:category" type="string" _value="Desktop &amp; Windows"/>
 	<oaf_attribute name="panel:icon" type="string" value="gnome-panel-window-list"/>
 	<oaf_attribute name="bugzilla:bugzilla" type="string" value="GNOME"/>
 	<oaf_attribute name="bugzilla:product" type="string" value="gnome-panel"/>
@@ -122,6 +125,7 @@
 	</oaf_attribute>
 	<oaf_attribute name="name" type="string" _value="Show Desktop"/>
 	<oaf_attribute name="description" type="string" _value="Hide application windows and show the desktop"/>
+	<oaf_attribute name="panel:category" type="string" _value="Desktop &amp; Windows"/>
 	<oaf_attribute name="panel:icon" type="string" value="gnome-fs-desktop.png"/>
 	<oaf_attribute name="bugzilla:bugzilla" type="string" value="GNOME"/>
 	<oaf_attribute name="bugzilla:product" type="string" value="gnome-panel"/>
