diff -Nur gnome-panel-2.13.91/applets/clock/clock.schemas.in gnome-panel-2.13.91.new/applets/clock/clock.schemas.in
--- gnome-panel-2.13.91/applets/clock/clock.schemas.in	2005-12-30 12:13:11.000000000 +0100
+++ gnome-panel-2.13.91.new/applets/clock/clock.schemas.in	2006-03-02 18:58:37.000000000 +0100
@@ -69,7 +69,7 @@
         <key>/schemas/apps/clock_applet/prefs/show_date</key>
         <owner>clock-applet</owner>
         <type>bool</type>
-        <default>false</default>
+        <default>true</default>
         <locale name="C">
           <short>Show date in clock</short>
           <long>If true, display date in the clock, in addition to time.</long>
