diff -Nur gnome-panel-2.15.92/applets/notification_area/GNOME_NotificationAreaApplet.xml gnome-panel-2.15.92.new/applets/notification_area/GNOME_NotificationAreaApplet.xml
--- gnome-panel-2.15.92/applets/notification_area/GNOME_NotificationAreaApplet.xml	2004-06-11 12:53:16.000000000 +0200
+++ gnome-panel-2.15.92.new/applets/notification_area/GNOME_NotificationAreaApplet.xml	2006-08-22 11:36:05.000000000 +0200
@@ -3,6 +3,9 @@
       <popup name="button3">
          <menuitem name="Notification Area Help Item" verb="SystemTrayHelp" _label="_Help"
                    pixtype="stock" pixname="gtk-help"/>
+          <separator/>
+             <placeholder name="LaunchpadItems"/>
+          <separator/>
          <menuitem name="Notification Area About Item" verb="SystemTrayAbout" _label="_About"
                    pixtype="stock" pixname="gnome-stock-about"/>
       </popup>
diff -Nur gnome-panel-2.15.92/applets/notification_area/main.c gnome-panel-2.15.92.new/applets/notification_area/main.c
--- gnome-panel-2.15.92/applets/notification_area/main.c	2006-08-15 16:39:05.000000000 +0200
+++ gnome-panel-2.15.92.new/applets/notification_area/main.c	2006-08-22 11:37:28.000000000 +0200
@@ -29,6 +29,7 @@
 #include <gtk/gtkmessagedialog.h>
 #include <gtk/gtkhbox.h>
 #include <libgnomeui/gnome-help.h>
+#include <lpint-bonobo.h>
 
 #include "na-tray-manager.h"
 #include "fixedtip.h"
@@ -580,6 +581,7 @@
   SystemTray *tray;
   AtkObject  *atko;
   int         screen_number;
+  BonoboUIComponent* popup_component;
   
   if (!(strcmp (iid, "OAFIID:GNOME_NotificationAreaApplet") == 0 ||
         strcmp (iid, "OAFIID:GNOME_SystemTrayApplet") == 0))
@@ -706,6 +708,10 @@
                                      NULL,
                                      menu_verbs,
                                      tray);
+
+  popup_component = panel_applet_get_popup_component (applet);
+
+  launchpad_integration_add_bonobo_ui(popup_component, "/popups/button3/LaunchpadItems");
   
   return TRUE;
 }
diff -Nur gnome-panel-2.15.92/applets/wncklet/GNOME_ShowDesktopApplet.xml gnome-panel-2.15.92.new/applets/wncklet/GNOME_ShowDesktopApplet.xml
--- gnome-panel-2.15.92/applets/wncklet/GNOME_ShowDesktopApplet.xml	2004-06-11 12:53:17.000000000 +0200
+++ gnome-panel-2.15.92.new/applets/wncklet/GNOME_ShowDesktopApplet.xml	2006-08-22 11:36:05.000000000 +0200
@@ -3,6 +3,9 @@
     <popup name="button3">
       <menuitem name="Show Desktop Help Item" verb="ShowDesktopHelp" _label="_Help"
     		    pixtype="stock" pixname="gtk-help"/>
+      <separator/>
+             <placeholder name="LaunchpadItems"/>
+      <separator/>
       <menuitem name="Show Desktop About Item" verb="ShowDesktopAbout" _label="_About"
     		    pixtype="stock" pixname="gnome-stock-about"/>
     </popup>
diff -Nur gnome-panel-2.15.92/applets/wncklet/GNOME_WorkspaceSwitcherApplet.xml gnome-panel-2.15.92.new/applets/wncklet/GNOME_WorkspaceSwitcherApplet.xml
--- gnome-panel-2.15.92/applets/wncklet/GNOME_WorkspaceSwitcherApplet.xml	2004-06-11 12:53:17.000000000 +0200
+++ gnome-panel-2.15.92.new/applets/wncklet/GNOME_WorkspaceSwitcherApplet.xml	2006-08-22 11:36:05.000000000 +0200
@@ -5,6 +5,9 @@
     		    pixtype="stock" pixname="gtk-properties"/>
       <menuitem name="Pager Help Item" verb="PagerHelp" _label="_Help"
     		    pixtype="stock" pixname="gtk-help"/>
+      <separator/>
+             <placeholder name="LaunchpadItems"/>
+      <separator/>
       <menuitem name="Pager About Item" verb="PagerAbout" _label="_About"
     		    pixtype="stock" pixname="gnome-stock-about"/>
     </popup>
diff -Nur gnome-panel-2.15.92/applets/wncklet/showdesktop.c gnome-panel-2.15.92.new/applets/wncklet/showdesktop.c
--- gnome-panel-2.15.92/applets/wncklet/showdesktop.c	2006-08-08 20:12:26.000000000 +0200
+++ gnome-panel-2.15.92.new/applets/wncklet/showdesktop.c	2006-08-22 11:36:05.000000000 +0200
@@ -43,6 +43,7 @@
 #include <string.h>
 
 #define TIMEOUT_ACTIVATE 1000
+#include <lpint-bonobo.h>
 
 
 typedef struct {
@@ -410,6 +411,7 @@
 {
         ShowDesktopData *sdd;
 	AtkObject       *atk_obj;
+        BonoboUIComponent* popup_component;
 
 	panel_applet_set_flags (applet, PANEL_APPLET_EXPAND_MINOR);
 
@@ -499,6 +501,10 @@
 			  G_CALLBACK (button_drag_leave),
 			  sdd);
 
+        popup_component = panel_applet_get_popup_component (sdd->applet);
+
+        launchpad_integration_add_bonobo_ui(popup_component, "/popups/button3/LaunchpadItems");
+        
   	gtk_widget_show_all (sdd->applet);
 
         return TRUE;
diff -Nur gnome-panel-2.15.92/applets/wncklet/workspace-switcher.c gnome-panel-2.15.92.new/applets/wncklet/workspace-switcher.c
--- gnome-panel-2.15.92/applets/wncklet/workspace-switcher.c	2006-08-08 01:49:19.000000000 +0200
+++ gnome-panel-2.15.92.new/applets/wncklet/workspace-switcher.c	2006-08-22 11:36:05.000000000 +0200
@@ -38,6 +38,7 @@
 #define NEVER_SENSITIVE "never_sensitive"
 #define NUM_WORKSPACES "/apps/metacity/general/num_workspaces"
 #define WORKSPACE_NAME "/apps/metacity/workspace_names/name_1"
+#include <lpint-bonobo.h>
 
 typedef struct {
 	GtkWidget *applet;
@@ -409,6 +410,7 @@
 	PagerData *pager;
 	GError *error;
 	gboolean display_names;
+        BonoboUIComponent* popup_component;
 	
 	panel_applet_add_preferences (applet, "/schemas/apps/workspace_switcher_applet/prefs", NULL);
 	
@@ -524,6 +526,11 @@
 					      NULL);
 	}
 
+        popup_component = panel_applet_get_popup_component (PANEL_APPLET (pager->applet));
+
+        launchpad_integration_add_bonobo_ui(popup_component, "/popups/button3/LaunchpadItems");        
+
+
 	return TRUE;
 }
 
diff -Nur gnome-panel-2.15.92/configure.in gnome-panel-2.15.92.new/configure.in
--- gnome-panel-2.15.92/configure.in	2006-08-22 00:26:25.000000000 +0200
+++ gnome-panel-2.15.92.new/configure.in	2006-08-22 11:36:05.000000000 +0200
@@ -62,7 +62,7 @@
 
 dnl pkg-config dependency checks
 
-PKG_CHECK_MODULES(PANEL, ORBit-2.0 >= $ORBIT_REQUIRED gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED gtk+-2.0 >= $GTK_REQUIRED libgnome-2.0 >= $LIBGNOME_REQUIRED libgnomeui-2.0 >= $LIBGNOMEUI_REQUIRED gnome-desktop-2.0 >= $LIBGNOME_DESKTOP_REQUIRED gnome-vfs-2.0 >= $GNOME_VFS_REQUIRED libglade-2.0 >= $LIBGLADE_REQUIRED gconf-2.0 >= $GCONF_REQUIRED libgnome-menu >= $LIBGNOME_MENU_REQUIRED dbus-glib-1 >= $DBUS_GLIB_REQUIRED)
+PKG_CHECK_MODULES(PANEL, ORBit-2.0 >= $ORBIT_REQUIRED gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED gtk+-2.0 >= $GTK_REQUIRED lpint-bonobo launchpad-integration libgnome-2.0 >= $LIBGNOME_REQUIRED libgnomeui-2.0 >= $LIBGNOMEUI_REQUIRED gnome-desktop-2.0 >= $LIBGNOME_DESKTOP_REQUIRED gnome-vfs-2.0 >= $GNOME_VFS_REQUIRED libglade-2.0 >= $LIBGLADE_REQUIRED gconf-2.0 >= $GCONF_REQUIRED libgnome-menu >= $LIBGNOME_MENU_REQUIRED dbus-glib-1 >= $DBUS_GLIB_REQUIRED)
 AC_SUBST(PANEL_CFLAGS)
 AC_SUBST(PANEL_LIBS)
 
@@ -70,7 +70,7 @@
 AC_SUBST(LIBPANEL_APPLET_CFLAGS)
 AC_SUBST(LIBPANEL_APPLET_LIBS)
 
-PKG_CHECK_MODULES(WNCKLET, gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED gtk+-2.0 >= $GTK_REQUIRED libgnomeui-2.0 >= $LIBGNOMEUI_REQUIRED libwnck-1.0 >= $LIBWNCK_REQUIRED libglade-2.0 >= $LIBGLADE_REQUIRED)
+PKG_CHECK_MODULES(WNCKLET, gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED gtk+-2.0 >= $GTK_REQUIRED lpint-bonobo launchpad-integration libgnomeui-2.0 >= $LIBGNOMEUI_REQUIRED libwnck-1.0 >= $LIBWNCK_REQUIRED libglade-2.0 >= $LIBGLADE_REQUIRED)
 AC_SUBST(WNCKLET_CFLAGS)
 AC_SUBST(WNCKLET_LIBS)
 
diff -Nur gnome-panel-2.15.92/gnome-panel/panel-context-menu.c gnome-panel-2.15.92.new/gnome-panel/panel-context-menu.c
--- gnome-panel-2.15.92/gnome-panel/panel-context-menu.c	2006-08-09 17:10:51.000000000 +0200
+++ gnome-panel-2.15.92.new/gnome-panel/panel-context-menu.c	2006-08-22 11:36:05.000000000 +0200
@@ -44,6 +44,7 @@
 #include "panel-properties-dialog.h"
 #include "panel-lockdown.h"
 #include "panel-addto.h"
+#include <launchpad-integration.h>
 
 static void
 panel_context_menu_show_help (GtkWidget *w,
@@ -257,6 +258,9 @@
 				  panel_profile_id_lists_are_writable ());
 
 	add_menu_separator (menu);
+
+       launchpad_integration_add_items (menu,-1,TRUE,FALSE);
+
 }
 
 GtkWidget *
